File: Java/Dado.java
package civitas;

import java.util.Random;

public class Dado {

    private Random random;
    private int ultimoResultado;
    private boolean debug;
    static final private Dado instance = new Dado();
    final private int SalidaCarcel = 5;
    
    private Dado() {
        ultimoResultado = -1;
        debug = false;
        random = new Random();
    }

    static public Dado getInstance() {
        return instance;
    }

    int tirar() {
        if (!debug) {
            ultimoResultado = random.nextInt(6) + 1;
            return ultimoResultado;
        } 
        else
            return 1;
    }
    
    boolean salgoDeLaCarcel(){
        boolean salir = false;
        
        if(tirar() >= SalidaCarcel)
            salir = true;
        return salir;    
    }
    
    int quienEmpieza(int n) {
        int jug = random.nextInt(n);
        
        return jug;
    }
    
    public void setDebug(boolean d) {
        debug = d;
        Diario.getInstance().ocurreEvento("Debug mode: " + debug);
    }
    
    int getUltimoResultado() {
        return ultimoResultado;
    }

}

File: Ruby/dado.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require 'singleton'
module Civitas
  class Dado
    include Singleton
    srand()
    attr_accessor = :ultimo_resultado
    def initialize
      @@salida_carcel = 5
      @ultimo_resultado = 0
      @debug = false
    end
    
    def tirar
      tirada = 1
      if (@debug == false)
        tirada = rand(6)+1.to_i
      end
      @ultimo_resultado = tirada
      return tirada
    end
    
    def salgo_de_la_carcel
      sale = false
      tirada = tirar()
      if (tirada >= 5)
        sale = true
      end
      return sale
    end
    
    def quien_empieza(n)
      numjugador = (rand(n) + 1).to_i
    end
    
    def ultimo_resultado
      @ultimo_resultado
    end
    
    def set_debug(d)
      @debug = d
      if (@debug == true)
        Diario.instance.ocurre_evento("Se ha activado el debug")
      else 
        if (@debug == false)
          Diario.instance.ocurre_evento("Se ha desactivado el debug")
        end
      end
    end
    
  end
end